wai-app-static/Network/Wai/Application/Static.hs

```diff
@@ -214,10 +214,10 @@ staticApp set req = staticAppPieces set (W.pathInfo req) req
 
 staticAppPieces :: StaticSettings -> [Text] -> W.Application
 staticAppPieces _ _ req sendResponse
-    | W.requestMethod req /= "GET" = sendResponse $ W.responseLBS
+    | notElem (W.requestMethod req) ["GET", "HEAD"] = sendResponse $ W.responseLBS
```

Comment:
Thanks for the note :) My haskell work has been limited to this point (my day job is mostly in [pick your favorite imperative language]), so any optimization tips are pretty useful to me.

In regards to your first comment, this PR is merging directly into the get-head branch and not master so that it could all go in as one feature. If you would prefer, I can reopen this as a PR to master (in which case I believe yes, we should wait for warp merge).